src/ngraph/node.hpp

```diff
@@ -142,7 +169,7 @@ namespace ngraph
         /// graph against the graph.
         bool is_same_op_type(const std::shared_ptr<Node>& node) const
         {
-            return description() == node->description();
+            return &description() == &node->description();
```

Comment:
I really doubt this change is necessary. I would be surprised if you could measure any performance difference and I would not be surprised if string compare did this internally.